src/lib.rs

```diff
@@ -1230,17 +1190,28 @@ impl Capture<Dead> {
 
 #[cfg(not(windows))]
 impl AsRawFd for Capture<Active> {
+    /// Returns the file descriptor for a live capture.
     fn as_raw_fd(&self) -> RawFd {
-        unsafe {
-            let fd = raw::pcap_fileno(*self.handle);
+        let fd = unsafe { raw::pcap_fileno(*self.handle) };
+        assert!(fd != -1, "Unable to get file descriptor for live capture");
+        fd
+    }
```

Comment:
why force your style ? some people see unsafe as an unit the scope was hinting that untill we check fd value for erreur the code was unsafe (thus I like the change for assert)

Reply:
Returning an invalid fd cannot cause UB otherwise `as_raw_fd` would have to be marked as `unsafe`, so only the function call is unsafe. But I do not mind changing it back.

Reply:
Nevermind, it's impossible since the device should be active in this implementation of as_raw_fd, so assert is what what want